R/FilterStateChoices.R

```diff
@@ -168,10 +168,11 @@ ChoicesFilterState <- R6::R6Class( # nolint
         x <- factor(as.character(x), levels = as.character(sort(unique(x))))
       }
       x <- droplevels(x)
-      choices <- table(x)
-      private$set_choices(names(choices))
-      self$set_selected(names(choices))
-      private$set_choices_counts(unname(choices))
+      choices_table <- table(x)
+      private$set_choices(names(choices_table))
+      self$set_selected(names(choices_table))
+      private$set_choices_counts(unname(choices_table))
+      private$set_is_choice_limited(x, choices)
```

Comment:
`choices` and `selected` argument are ignored. Constructor sets is_choice_limited without limiting anything

```r
state <- ChoicesFilterState$new(
  letters,
  x_reactive = reactive(NULL),
  "test",
  "var",
  choices = c("a", "b"),
  selected = c("a", "b"),
)

shiny::isolate(state$get_state())
# TRUE
state$.__enclos_env__$private$is_choice_limited
# returns all letters and everything is selected
```